pkg/distributor/push_test.go

```diff
@@ -846,6 +845,114 @@ func TestHandler_ErrorTranslation(t *testing.T) {
 		})
 	}
 }
+func TestHandler_HandleRetryAfterHeader(t *testing.T) {
+	const (
+		DefaultRetryBase      = 3 * time.Second
+		DefaultMaxAllowedAtts = 2
+	)
+
+	testCases := []struct {
+		name           string
+		parserError    bool
+		handlerError   error
```

Comment:
this may be just me, but this test also relies on having the logic for calculating response codes unchanged. Which makes it more brittle and we don't really care about how errors are mapped when we test for the Retry-After header. 

Should we instead rewrite this test to only test how `addHeaders` works? This would also remove the need to worry about an end-to-end HTTP request test (which IMO is a benefit)